src/checkbox/src/components/Checkbox.js

```diff
@@ -0,0 +1,84 @@
+// @flow
+import React, { Component } from 'react';
+import { CheckboxIndicator, CheckboxInput, CheckboxLabel } from './styled';
+
+type Props = {
+  disabled ? :boolean;
+  onChange ? :(event :SyntheticInputEvent<HTMLInputElement>) => mixed;
+  value ? :any;
+  label ? :string;
+  name ? :string;
+  checked ? :boolean;
+  defaultChecked ? :boolean;
+}
+
+type State = {
+  isChecked :boolean;
+};
+
+class Checkbox extends Component<Props, State> {
+
+  static defaultProps = {
+    checked: undefined,
+    defaultChecked: false,
+    disabled: false,
+    label: undefined,
+    name: undefined,
+    onChange: undefined,
+    value: undefined,
+  };
+
+  constructor(props :Props) {
+    super(props);
+
+    const { checked, defaultChecked } = props;
+    this.state = {
+      isChecked: (checked !== undefined)
+        ? checked
+        // $FlowFixMe optional prop provided by defaultProps
+        : defaultChecked,
+    };
+  }
+
+  // Use prop 'checked' over state 'isChecked' if available
+  getIsChecked = () => {
+    const { checked } = this.props;
+    const { isChecked } = this.state;
+    return checked !== undefined ? checked : isChecked;
+  }
+
+  handleChange = (e :SyntheticInputEvent<HTMLInputElement>) => {
+    const { disabled, onChange } = this.props;
+    e.persist();
```

Comment:
What is `persist()`?

Reply:
https://reactjs.org/docs/events.html#event-pooling

In case the event should be used asynchronously, the pooled SyntheticEvent will not be reused by React. It's probably not super necessary just something I stumbled across when looking into SyntheticEvents.

Reply:
I see. So why are you passing the event to the `onChange()` handler? My expectation as a user would be to receive the checked / not checked boolean in the callback.

Reply:
I felt if the purpose of LUK was to provide a base unit for components, they should behave as close to native as possible. Including being able to distinguish the name and value of the target and all other things possible with EventTarget. An example of this is in `CheckboxController`